smartsim/settings/launch_settings.py

```diff
@@ -165,6 +174,15 @@ def env_vars(self, value: dict[str, str | None]) -> None:
 
         :param value: The new environment mapping
         """
+        if not (
+            isinstance(value, t.Mapping)
+            and all(
```

Comment:
Similar discrepancy here: type hint is looking for a `dict` specifically, but runtime check is for a `Mapping`

Reply:
changed